File: web/src/api/lg/revoke.js
import service from '@/utils/request'

export const createRevoke = (data) => {
  return service({
    url: '/revoke/createRevoke',
    method: 'post',
    data
  })
}

export const deleteRevoke = (data) => {
  return service({
    url: '/revoke/deleteRevoke',
    method: 'delete',
    data
  })
}

export const deleteRevokeByIds = (data) => {
  return service({
    url: '/revoke/deleteRevokeByIds',
    method: 'delete',
    data
  })
}

export const updateRevoke = (data) => {
  return service({
    url: '/revoke/updateRevoke',
    method: 'put',
    data
  })
}

export const findRevoke = (params) => {
  return service({
    url: '/revoke/findRevoke',
    method: 'get',
    params
  })
}

export const getRevokeList = (params) => {
  return service({
    url: '/revoke/getRevokeList',
    method: 'get',
    params
  })
}

File: server/router/lg/revoke.go
package lg

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type RevokeRouter struct {
}

func (s *RevokeRouter) InitRevokeRouter(Router *gin.RouterGroup) {
	revokeRouter := Router.Group("revoke").Use(middleware.OperationRecord())
	revokeRouterWithoutRecord := Router.Group("revoke")
	var revokeApi = v1.ApiGroupApp.LgApiGroup.RevokeApi
	{
		revokeRouter.POST("createRevoke", revokeApi.CreateRevoke)             // 新建Revoke
		revokeRouter.DELETE("deleteRevoke", revokeApi.DeleteRevoke)           // 删除Revoke
		revokeRouter.DELETE("deleteRevokeByIds", revokeApi.DeleteRevokeByIds) // 批量删除Revoke
		revokeRouter.PUT("updateRevoke", revokeApi.UpdateRevoke)              // 更新Revoke
	}
	{
		revokeRouterWithoutRecord.GET("findRevoke", revokeApi.FindRevoke)       // 根据ID获取Revoke
		revokeRouterWithoutRecord.GET("getRevokeList", revokeApi.GetRevokeList) // 获取Revoke列表
	}
}
